src/workers/parser/getOutOfScopeLocations.js

```diff
@@ -38,26 +40,62 @@ function getLocation(func) {
 }
 
 /**
- * Reduces an array of locations to remove items that are completely enclosed
- * by another location in the array.
+ * Find the nearest location containing the input position and
+ * return new locations without inner locations under that nearest location
+ *
+ * @param locations Notice! The locations MUST be sorted by `sortByStart`
+ *                  so that we can do linear time complexity operation.
  */
-function removeOverlaps(
-  locations: AstLocation | AstLocation[],
-  location: AstLocation
-) {
-  // support reducing without an initializing array
-  if (!Array.isArray(locations)) {
-    locations = [locations];
+function removeInnerLocations(locations: AstLocation[], position: AstPosition) {
+  // First, let's find the nearest position-enclosing function location,
+  // which is to loop locations to find the last location enclosing the position.
+  const newLocs = locations.slice();
```

Comment:
The reason returning new locations is because:
1. Both `removeOverlaps` and Array `filter` return new locations. Since `removeInnerLocations` also does removal operation, that is for consistency, not to bring surprise. 
2. Although this is not an exported function, for the removal operation probably less surprise without mutating the input arg. This is not necessary, however, consider the reason 1, consistency is good for less error.